scripts/write_lh_annotations.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import math
+import copy
+import argparse
+import csv
+import os
+import sys
+default_partis_path = os.path.join(os.getcwd(), 'lib/partis')
+sys.path.append(os.path.join(default_partis_path, 'python'))
+import utils
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Combine original partis_run.yaml and linearham_run.log to update reflect linearham annotations")
+    parser.add_argument(
+        "partis_yaml_file", type=str,
+        help="Path to partis output YAML file.")
+    parser.add_argument(
+        "linearham_log_file", type=str,
+        help="Path to linearham_run.log TSV file.")
+    parser.add_argument(
+        "--output-base", type=str, required=True,
+        help="The base  output path for linearham annotations.")
+    parser.add_argument(
+        "--collapse-annotations-by",
+        type=lambda x: x.split(','), help='comma separated list of annotation keys by which we should define a unique annotation', default=None)
+
+    def read_linearham_lines(lh_annotation_tsv):
+        with open(lh_annotation_tsv) as tsvfile:
+            return [l for l in csv.DictReader(tsvfile, dialect='excel-tab')]
+            
+    def match(d1, d2, keys_to_match):
```

Comment:
suggestion:

return all(d1[k] == d2[k] for k in keys_to_match)

maybe not any faster, but it's more pythonic I think.